proto/internal/temporal/server/api/matchingservice/v1/request_response.proto

```diff
@@ -113,6 +113,8 @@ message AddWorkflowTaskRequest {
     // If using build-id based versioning: version stamp of last worker to complete a workflow
     // task for this workflow. Used to dispatch to correct worker in matching.
     temporal.api.common.v1.WorkerVersionStamp worker_version_stamp = 10;
+    // True if this is the first workflow task for the workflow.
+    bool is_first_workflow_task = 11;
```

Comment:
Would it make more sense to wrap these 2 fields in another proto message so it's clear this is all for versioning and to avoid the sprawl of versioning related fields?
You might also want to expressly state in history that you want to use the unversioned queue instead of switching behavior based on `is_first_workflow_task`.

Reply:
Looks like we're thinking in the same direction, I added a comment elsewhere.